modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ScriptProcessor.java

```diff
@@ -121,7 +143,12 @@ public ScriptProcessor create(Map<String, Processor.Factory> registry, String pr
                 throw newConfigurationException(TYPE, processorTag, null, "Could not initialize script");
             }
 
-            return new ScriptProcessor(processorTag, script, scriptService);
+            CompiledScript compiledInlineScript = null;
+            if (script.getType() == ScriptType.INLINE) {
+                compiledInlineScript = scriptService.compile(script, ScriptContext.Standard.INGEST, script.getOptions());
```

Comment:
I don't think the compile script should be stashed here. Instead, just check that the script can be resolved with the script service (don't just check inline; for example, it would then also ensure if they use a file script it exists).